runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -4224,13 +4229,18 @@ TR_MultipleCallTargetInliner::exceedsSizeThreshold(TR_CallSite *callSite, int by
      // HACK: Get frequency from both sources, and use both.  You're
      // only cold if you're cold according to both.
 
+     bool isLambdaFormGeneratedMethod = comp()->fej9()->isLambdaFormGeneratedMethod(callerResolvedMethod);
+     // TODO: we should ignore frequency for thunk archetype, however, this require performance evaluation
+     bool frequencyIsInaccurate = isLambdaFormGeneratedMethod;
```

Comment:
Could you please explain this further ? i.e. why is the frequency inaccurate for lambda form methods ?

Reply:
LambdaForm in a MethodHandle can change. The change usually happen when customization happens, i.e. when a MethodHandle has been executed frequent enough, a customized LambdaForm will replace the existing LambdaForm. The frequency information of the original LambdaForm is not inherited to the customized one, thus the customized one might seem cold when it's compiled or inlined.